nntrainer/src/neuralnet.cpp

```diff
@@ -124,118 +124,137 @@ int NeuralNetwork::setConfig(std::string config) {
+  section_names_iter =
+    std::find(section_names.begin(), section_names.end(), "network");
+  if (section_names_iter == section_names.end()) {
+    ml_loge("Error: Network section not found in the .");
+    return ML_ERROR_INVALID_PARAMETER;
+  } else {
+    section_names.erase(section_names_iter);
+  }
+
+  net_type = (nntrainer::NetType)parseType(
+    iniparser_getstring(ini, "Network:Type", unknown), TOKEN_NET);
   learning_rate = iniparser_getdouble(ini, "Network:Learning_rate", 0.0);
   decay_rate = iniparser_getdouble(ini, "Network:Decay_rate", 0.0);
   decay_steps = iniparser_getint(ini, "Network:Decay_steps", -1);
-
-  popt.learning_rate = learning_rate;
-  popt.decay_steps = decay_steps;
-  popt.decay_rate = decay_rate;
   epoch = iniparser_getint(ini, "Network:Epoch", 100);
-  status = opt.setType((OptType)parseType(
-    iniparser_getstring(ini, "Network:Optimizer", unknown), TOKEN_OPT));
-  NN_INI_RETURN_STATUS();
-
   cost = (CostType)parseType(iniparser_getstring(ini, "Network:Cost", unknown),
                              TOKEN_COST);
   model = iniparser_getstring(ini, "Network:Model", model_name);
   batch_size = iniparser_getint(ini, "Network:Minibatch", 1);
 
+  status = opt.setType((OptType)parseType(
+    iniparser_getstring(ini, "Network:Optimizer", unknown), TOKEN_OPT));
+  NN_INI_RETURN_STATUS();
+  popt.learning_rate = learning_rate;
+  popt.decay_steps = decay_steps;
+  popt.decay_rate = decay_rate;
   popt.beta1 = iniparser_getdouble(ini, "Network:beta1", 0.0);
   popt.beta2 = iniparser_getdouble(ini, "Network:beta2", 0.0);
   popt.epsilon = iniparser_getdouble(ini, "Network:epsilon", 0.0);
 
   status = opt.setOptParam(popt);
   NN_INI_RETURN_STATUS();
 
-  for (unsigned int i = 0; i < layers_name.size(); i++)
-    ml_logi("%s", layers_name[i].c_str());
-
-  loss = 100000.0;
-
-  if (iniparser_find_entry(ini, "DataSet:TrainData")) {
-    data_buffer = std::make_shared<DataBufferFromDataFile>();
-
-    std::shared_ptr<DataBufferFromDataFile> dbuffer =
-      std::static_pointer_cast<DataBufferFromDataFile>(data_buffer);
-
-    status = dbuffer->setDataFile(
-      iniparser_getstring(ini, "DataSet:TrainData", ""), DATA_TRAIN);
-    NN_INI_RETURN_STATUS();
-    status = dbuffer->setDataFile(
-      iniparser_getstring(ini, "DataSet:ValidData", ""), DATA_VAL);
-    NN_INI_RETURN_STATUS();
-    status = dbuffer->setDataFile(
-      iniparser_getstring(ini, "DataSet:TestData", ""), DATA_TEST);
-    NN_INI_RETURN_STATUS();
-    status = dbuffer->setDataFile(
-      iniparser_getstring(ini, "DataSet:LabelData", ""), DATA_LABEL);
-    NN_INI_RETURN_STATUS();
-
-  } else if (iniparser_find_entry(ini, "DataSet:Tflite")) {
-    ml_loge("Error: Not yet implemented!");
-    return ML_ERROR_INVALID_PARAMETER;
-  } else {
-    data_buffer = std::make_shared<DataBufferFromCallback>();
+  /** Parse the DataSet section */
+  data_buffer = std::make_shared<DataBufferFromCallback>();
+  section_names_iter =
```

Comment:
This 203 line has conflict with line 212

Reply:
Yes, this has been updated.